generic_tracers/generic_COBALT.F90

```diff
@@ -4313,9 +4321,40 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
 
           ! Iron from sediment (Dale, 2015).  The maximum release from the sediment is set by ffe_sed_max.  The
           ! hyperbolic tangent requires the flux of carbon to the sediments (as mmoles m-2 day-1) in the numerator
-          ! and the bottom water oxygen concentration (in microMolar units) in the denominator:
+          ! and the bottom water oxygen concentration (in microMolar units) in the denominator. Note that ffe_sed_max
+          ! was converted to moles Fe m-2 sec-1 during parameter input, so ffe_sed has is in moles Fe m-2 sec-1
           cobalt%ffe_sed(i,j) = cobalt%ffe_sed_max * tanh( (cobalt%fntot_btm(i,j)*cobalt%c_2_n*sperd*1.0e3)/ &
                                 max(cobalt%btm_o2(i,j)*1.0e6,epsln) )
+
+          ! Additional coastal iron (Optional, default fe_coast = 0)
+          !
+          ! Coarse resolution models and/or intermediate resolution models in areas with exceptionally steep bathymetry
+          ! can under-represent coastal iron because they don't resolve shallow regions. An option to add iron through
+          ! the vertical face of the land mass has thus been included.  The flux is posed as a fraction (fe_coast) of
+          ! the sediment Fe flux (moles Fe m-2 sec-1) that would have result from the sinking organic matter flux and
+          ! O2 level of the adjacent waters.  This then spread across the layer mass (rho_dzt(i,j,k)) to give an input
+          ! in moles Fe kg-1 sec-1. Conceptually, this can be though of as a net iron flux resulting from the fraction
+          ! of the sinking flux that would have been intercepted at shallower depths were the model resolution finer.
+          ! default value of fe_coast is 0 (i.e., only the explicitly resolved benthic flux is included).
+          !
+          ! Old Expression:
+          ! cobalt%jfe_coast(i,j,1) = cobalt%fe_coast * mask_coast(i,j) * grid_tmask(i,j,1) / &
+          !     sqrt(grid_dat(i,j))
+          !
+          do k = 1, nk !{
```

Comment:
Hi @charliestock,
I think I found the issue. At line 4241, we specify `k = grid_kmt(i,j)` to ensure that we use the model bottom later for the calculation in the sediment section.
https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/07b96471c2868a313ad4d56baa6f627e002d604e/generic_tracers/generic_COBALT.F90#L4241
However, the new code changes the k value and eventually affects `fcased_burial` starting at line 4418:
https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/07b96471c2868a313ad4d56baa6f627e002d604e/generic_tracers/generic_COBALT.F90#L4418C22-L4427

The solution would be to add `k = grid_kmt(i,j)` either right after line 4357 or right before line 4418, or just simply change line4426 to `grid_tmask(i,j,grid_kmt(i,j))`

Reply:
I agree that this is likely the issue.  We should, however, discuss the role of grid_kmt in MOM6 when we meet tomorrow.  My understanding is that grid_kmt = nk everywhere for MOM6, so I'm not sure this oversight actually changed anything.  Let's implement and find out!

Reply:
@charliestock, I think the main issue is that in Fortran after 'k=1,nk', the k value will be +1 over nk (76) in memory, which will be out of bound and then causing potential issue. But sure we can discuss more tomorrow.